scripts/validate_docstrings.py

```diff
@@ -690,8 +691,14 @@ def get_validation_data(doc):
                 errs.append(error('PR09', param_name=param))
 
     if doc.is_function_or_method:
+        (tree, ) = ast.parse(doc.method_source.strip()).body
         if not doc.returns:
-            if 'return' in doc.method_source:
+            returns = [node.value for node in ast.walk(tree)
+                       if isinstance(node, ast.Return)]
+            non_bare_returns = [r for r in returns if r is not None and
+                                not (isinstance(r, ast.NameConstant) and
+                                     r.value is None)]
+            if any(non_bare_returns):
```

Comment:
Sorry, I think this change comes from a comment from @WillAyd, but we can't have all this code here, we need to encapsulate this in a method.

Also, when you do that, the second comprehension, should be a loop, it's too complex.

Other than that, this looks much better, good work.

Reply:
Yea this is pretty tough to read. Comments may be helpful as well

Reply:
I have encapsulated this in a property.